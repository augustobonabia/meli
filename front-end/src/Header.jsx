import React from 'react';
import SearchBar from './SearchBar';
import './Header.scss';

function Header() {
  return (
    <header className="header">
      <div className="page-container">
        <div className="logo-container">
          <img className="logo" alt="logo" src="../logo.png" />
        </div>
        <SearchBar />
      </div>
    </header>
  );
}

export default Header;
