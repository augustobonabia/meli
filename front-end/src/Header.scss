@import "variables.scss";
@import "mixins.scss";

.header {
  @include aligned-grow-flex;
  background-color: $app-color;
  padding: $tiny-space 0;

  div.logo-container {
    @include aligned-flex;
    margin-right: $small-space;

    img.logo {
      cursor: pointer;
      height: $input-font-size + 2 * $input-padding;
    }
  }

  form.search-form {
    display: inline-flex;
    max-width: 1200px;
    flex-grow: 1;
    flex-wrap: nowrap;
    justify-content: center;

    input[type="text"].search-box {
      flex-grow: 1;
      border-radius: $border-radious 0 0 $border-radious;
    }

    button.search-button {
      padding: 0 10px;
      display: flex;
      background-color: $page-background-color;
      align-items: center;
      justify-content: center;
      border-radius: 0 $border-radious $border-radious 0;

      img {
        height: $input-font-size * 0.75;
      }
    }
  }
}
