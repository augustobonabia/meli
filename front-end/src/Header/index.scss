@import "../core/variables.scss";
@import "../core/mixins.scss";

.header {
  background-color: $app-color;
  padding: $tiny-space 0;

  div.logo-container {
    @include aligned-flex;
    margin-right: $small-space;

    a {
      @include aligned-flex;
      img.logo {
        cursor: pointer;
        height: $input-font-size + 2 * $input-padding;
      }
    }
  }

  form.search-form {
    display: inline-flex;
    flex-grow: 1;
    flex-wrap: nowrap;
    justify-content: center;

    input[type="text"].search-box {
      flex-grow: 1;
      border-radius: $border-radius 0 0 $border-radius;
    }

    button.search-button {
      padding: 0 10px;
      display: flex;
      background-color: $page-background-color;
      align-items: center;
      justify-content: center;
      border-radius: 0 $border-radius $border-radius 0;

      img {
        height: $input-font-size * 0.75;
      }
    }
  }
}
