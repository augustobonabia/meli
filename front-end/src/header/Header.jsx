import React from 'react';
import { Link } from 'react-router-dom';
import SearchBar from './SearchBar';
import './header.scss';

function Header() {
  return (
    <header className="header">
      <div className="page-container">
        <div className="logo-container">
          <Link to="/">
            <img className="logo" alt="logo" src="../logo.png" />
          </Link>
        </div>
        <SearchBar />
      </div>
    </header>
  );
}

export default Header;
