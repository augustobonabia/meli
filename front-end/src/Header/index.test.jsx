import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './index';
import renderer from 'react-test-renderer';

it('El header se renderiza correctamente', () => {
  const tree = renderer
    .create(<Router><Header /></Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
