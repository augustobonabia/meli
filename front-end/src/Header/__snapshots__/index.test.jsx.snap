// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El header se renderiza correctamente 1`] = `
<header
  className="page-section header"
>
  <div
    className="page-section-container"
  >
    <div
      className="logo-container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="logo"
          src="../logo.png"
        />
      </a>
    </div>
    <form
      className="search-form"
      name="searchForm"
      onSubmit={[Function]}
    >
      <input
        aria-label="Ingresá tu búsqueda"
        className="search-box"
        name="search"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar"
        size="1"
        type="text"
        value=""
      />
      <button
        className="search-button"
        type="submit"
        value="Submit"
      >
        <img
          alt="searchIcon"
          src="./search.png"
        />
      </button>
    </form>
  </div>
</header>
`;
